lib/macho/load_commands.rb

```diff
@@ -204,25 +204,25 @@ def to_s
     # pretend that strings stored in LCs are immediately available without
     # explicit operations on the raw Mach-O data.
     class LCStr
-      # @param raw_data [String] the raw Mach-O data.
       # @param lc [MachO::LoadCommand] the load command
       # @param lc_str [Fixnum] the offset to the beginning of the string
       # @api private
-      def initialize(raw_data, lc, lc_str)
-        @raw_data = raw_data
-        @lc = lc
-        @lc_str = lc_str
-        @str = @raw_data.slice(@lc.offset + @lc_str...@lc.offset + @lc.cmdsize).delete("\x00")
+      def initialize(lc, lc_str)
+        raw_data = lc.view.raw_data
+        lc_str_abs = lc.offset + lc_str
+        lc_end = lc.offset + lc.cmdsize - 1
```

Comment:
Upon closer inspection, I wonder if this line and the line above this one should directly use `lc.view.offset` instead of relying on the deprecated `LoadCommand#offset` method.